src/components/SlideDeck/SlideDeck.js

```diff
@@ -1,52 +1,106 @@
+import {
+    Fullscreen,
+    KeyboardArrowLeft,
+    KeyboardArrowRight,
+} from "@material-ui/icons";
 
 class Slide extends React.Component {
     static propTypes = {
         sticky: PropTypes.bool,
         stickyUntil: PropTypes.number,
-    }
+    };
 
-    constructor(props){
+    constructor(props) {
         super(props);
         this.props = props;
     }
     render() {
-        return <Container className={`slide-content ${this.props.className}`}>{this.props.children}</Container>;
+        return (
+            <Container className={`slide-content ${this.props.className}`}>
+                {this.props.children}
+            </Container>
+        );
     }
 }
 
-class InvalidChildComponentError extends TypeError {}
+class InvalidChildComponentError extends TypeError { }
 
 class SlideDeck extends React.Component {
     static contextType = UxContext;
 
-    constructor(props){
+    constructor(props) {
         super(props);
         this.state = {
             currentSlide: 0,
+            isFullscreen: false,
             stickied: {
                 current: null,
                 previous: [],
             },
-         };
+        };
 
-        this.props.children.forEach(child => {
-            if(child.type.name !== "Slide" && process.env.NODE_ENV !== "production"){
-                throw new InvalidChildComponentError(`All children of SlideDeck must be Slide components. Got "${child.type.name}" instead`);
+        this.props.children.forEach((child) => {
+            if (
+                child.type.name !== "Slide" &&
+                process.env.NODE_ENV !== "production"
+            ) {
+                throw new InvalidChildComponentError(
+                    `All children of SlideDeck must be Slide components. Got "${child.type.name}" instead`
+                );
             }
-        })
+        });
+
+        this.nextSlide = this.nextSlide.bind(this);
+        this.prevSlide = this.prevSlide.bind(this);
+        this.handleSlideChange = this.handleSlideChange.bind(this);
+        this.handleKeyPress = this.handleKeyPress.bind(this);
+    }
+
+    componentDidMount() {
+        document.addEventListener("keydown", this.handleKeyPress, false);
+    }
+
+    componentWillUnmount() {
+        document.removeEventListener("keydown", this.handleKeyPress, false);
+    }
+
+    handleSlideChange(e) {
+        e.stopPropagation();
+        this.setState({
+            currentSlide: e.target.value,
+        });
+    }
 
-        this.handleClick = this.handleClick.bind(this);
-        this.handleOnContext = this.handleOnContext.bind(this);
+    handleKeyPress(e) {
+        if (e.key === "Escape") {
+            this.setState({
+                isFullscreen: false,
+            });
+        }
+        if (e.key === "ArrowLeft") {
+            this.prevSlide();
+        }
+        if (e.key === "ArrowRight") {
+            this.nextSlide();
+        }
     }
 
-    nextSlide() {
+    nextSlide(e) {
```

Comment:
`nextSlide` and `prevSlide` shouldn't require a parameter

Reply:
The reason it takes an optional event parameter is so that in the onClick handler, we can handle the case of event propagation without repeating it each time we call it. So by not including this, we have a pattern of `onClick={e => { e.stopPropagation(); this.nextSlide() }` instead of `onClick={this.nextSlide}`.

Reply:
I know, I'm saying to move the `e.StopPropagation()` outside of `nextSlide` and `prevSlide`. If `e` is undefined, then the function `StopPropagation` is undefined, which will throw an error and fail. Therefore, `e` is not an optional parameter.